### PR TITLE
[open62541] Add 1.2.2 version

### PR DESCRIPTION
**Pull request by Pro**
Specify library name and version:  **open62541/1.2.2**

Add new version 1.2.2 for open62541

ping @SpaceIm 

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/open62541/all/conanfile.py [hunk 1]
```diff
@@ -31,7 +31,7 @@ class Open62541Conan(ConanFile):
         "shared": [True, False],
         "historize": [True, False, "Experimental"],
         "logging_level": ["Fatal", "Error", "Warning", "Info", "Debug", "Trace"],
-        "subscription": [True, False, "With Events"],
+        "subscription": [True, False, "With Events", "With Events, Alarms, Conditions"],
```
Comment (madebr) [inline recipes/open62541/all/conanfile.py:34]: Those are bad option values.
`With Events` has spaces and `With Events, Alarms, Conditions` has space+commas.
How about the following?
```python
        "subscription": [True, False, "events", "events+alarms+conditions"],
```
You can supper the old `With Events` by adding it to the list, and doing
```
def configure(self):
    if self.options.subscription == "With Events":
        self.output.warning("`{name}:subscription=With Events` is deprecated. Use `{name}:subscription=events` instead".format(name=self.name))  # Deprecated in 1.2.2
```
(and use `events`+`events+alarms+conditions` instead in `_configure_cmake`)

Comment (Pro) [reply recipes/open62541/all/conanfile.py]: Changed as suggested. Thanks!

File: recipes/open62541/all/conanfile.py [hunk 2]
```diff
@@ -101,6 +102,9 @@ def configure(self):
             del self.settings.compiler.cppstd
             del self.settings.compiler.libcxx
 
+        if self.options.subscription == "With Events":
+            self.output.warning("`{name}:subscription=With Events` is deprecated. Use `{name}:subscription=events` instead".format(name=self.name))  # Deprecated in 1.2.2
```
Comment (madebr) [inline recipes/open62541/all/conanfile.py:106]: I forgot to do this.
```suggestion
            self.output.warning("`{name}:subscription=With Events` is deprecated. Use `{name}:subscription=events` instead".format(name=self.name))  # Deprecated in 1.2.2
            self.options.subscription = "events"
```

File: recipes/open62541/all/conanfile.py [hunk 3]
```diff
@@ -31,7 +31,8 @@ class Open62541Conan(ConanFile):
         "shared": [True, False],
         "historize": [True, False, "Experimental"],
         "logging_level": ["Fatal", "Error", "Warning", "Info", "Debug", "Trace"],
-        "subscription": [True, False, "With Events"],
+        # "With Events" is deprecated
+        "subscription": [True, False, "With Events", "events", "events+alarms+conditions"],
```
Comment (madebr) [inline recipes/open62541/all/conanfile.py:35]: When using commas (but without spaces in between), this makes this option the same interace as the `package` option of msys2.
```suggestion
        "subscription": [True, False, "With Events", "events", "alarms,conditions,events"],
```

**Comment by prince-chrismc** [comment]
It's nearly perfect

File: recipes/open62541/all/conanfile.py [hunk 4]
```diff
@@ -31,7 +31,8 @@ class Open62541Conan(ConanFile):
         "shared": [True, False],
         "historize": [True, False, "Experimental"],
         "logging_level": ["Fatal", "Error", "Warning", "Info", "Debug", "Trace"],
-        "subscription": [True, False, "With Events"],
+        # "With Events" is deprecated
+        "subscription": [True, False, "With Events", "events", "alarms,conditions,events"],
```
[the remark this answers is not in the document]
Comment (Pro) [reply recipes/open62541/all/conanfile.py:42]: Done!

**Review by prince-chrismc — APPROVED** [review]
Thanks for going the extra mile and comment all the options!

**Review by madebr — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**